kustomize/commands/internal/util/util.go

```diff
@@ -96,3 +96,19 @@ func trimQuotes(s string) string {
 	}
 	return s
 }
+
+// Removes paths containing hidden files/folders from a list of paths
+func RemoveHiddenFilesAndDirs(paths []string) []string {
+	if len(paths) == 0 {
+		return paths
+	}
+	var result []string
+	for _, path := range paths {
+		if strings.Contains(path, "/.") {
```

Comment:
A few questions:

1. Will this still work on windows?

2. What about hidden files in the current directory, e.g. if someone does `kustomize edit add resource *` and there is a hidden file in my current directory, e.g. `.github`, in which case "/." doesn't appear?

Suggestion: Change this check to `strings.HasPrefix(filepath.Base(path), ".")`

Reply:
@natasha41575 
I'm unable to understand one thing. If I have a directory which has the structure as follows:
```
dir/some-file-1
dir/some-file-2
dir/dir2/some-file3
```
`GlobPatterns` returns only `some-file-1`, `some-file-2` for the pattern `dir/*`. It does not go into nested directories, but the command `ls dir/*` does return the file in `dir2` on bash if I replicate the same directories on storage. Is the behavior of `GlobPatterns` correct? Should it not reference the nested directories (and hidden nested directories)?

Reply:
Deep down in implementation of `GlobPatterns`, kustomize does the following:

```
https://github.com/kubernetes-sigs/kustomize/blob/2b8a39373e880eb8d8f55d3dce090035f0ccc21c/kyaml/filesys/fsondisk.go#L90-94
```

Reply:
Deep down in the implementation of `GlobPatterns`, kustomize does the following:

https://github.com/kubernetes-sigs/kustomize/blob/2b8a39373e880eb8d8f55d3dce090035f0ccc21c/kyaml/filesys/fsondisk.go#L89-L92

IMO I would think that `filepath.Glob` does the "correct" globbing behavior, because it is a builtin Go package. In any case, I think kustomize should follow whatever `filepath.Glob` does rather than attempting to define its own correct behavior.

Reply:
Yes that makes sense

Reply:
On a second thought @natasha41575, I think correcting the globbing behavior of `filepath.Glob` is exactly what we're trying to do here (by explicitly removing files that `filepath.Glob` returns). I guess it really comes down to what would you expect if you had the above mentioned directory structure and used `dir/*` as the glob pattern (or, even just `*`). If you expect the file in `dir2` to be included and it isn't, it will be reported as a bug in the future, even though that is the standard behaviour of `filepath.Glob`. I don't use globbing enough to justify if that is correct or incorrect, but maybe someone who knows better (like yourself, or @knverey or @khrisrichardson) could decide if this should be a separate issue that needs to be fixed.

Reply:
> On a second thought @natasha41575, I think correcting the globbing behavior of filepath.Glob is exactly what we're trying to do here (by explicitly removing files that filepath.Glob returns). I guess it really comes down to what would you expect if you had the above mentioned directory structure and used dir/* as the glob pattern (or, even just *).

That's a fair point, and worth some more investigation. I also don't use globbing enough to be able to confidently state what is and what isn't correct behavior, but FWIW some google search results (https://stackoverflow.com/questions/1690809/what-expands-to-all-files-in-current-directory-recursively, https://unix.stackexchange.com/questions/49913/recursive-glob) seem to suggest that you need to do something special (`**`) for recursive globbing.